connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -0,0 +1,166 @@
+import { ModelId } from "@dust-tt/types";
+import { CheerioCrawler, Configuration } from "crawlee";
+import turndown from "turndown";
+
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertToDatasource } from "@connectors/lib/data_sources";
+import { Connector } from "@connectors/lib/models";
+import {
+  WebCrawlerConfiguration,
+  WebCrawlerFolder,
+} from "@connectors/lib/models/webcrawler";
+import {
+  reportInitialSyncProgress,
+  syncSucceeded,
+} from "@connectors/lib/sync_status";
+
+export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found.`);
+  }
+  const webCrawlerConfig = await WebCrawlerConfiguration.findOne({
+    where: {
+      connectorId,
+    },
+  });
+  if (!webCrawlerConfig) {
+    throw new Error(`Webcrawler configuration not found for connector.`);
+  }
+
+  return crawlWebsite(webCrawlerConfig.id);
+}
+
+export async function crawlWebsite(
+  webcrawlerConfigurationId: ModelId
```

Comment:
Let's not pass modelIds here. For now if we have a 1-1 relationship let's just merge the two functions?